File: frontend/src/types/match.ts
export type MatchSession = {
  id: number
  client_user_id: number
  status: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export type MatchMessage = {
  id: number
  session_id: number
  role: 'user' | 'assistant' | string
  content: string
  created_at: string
}

export type TherapistSummary = {
  user_id: number
  full_name: string
  specialization: string
  languages: string[] | string | null
  city: string | null
  is_online: boolean
  price_per_session: number | string | null
}

export type MatchResultPayload = {
  top5?: TherapistSummary[]
  reasons?: string[]
  extraction?: Record<string, unknown>

  /** Some workflows may include this inside the payload */
  assistant_message?: string
}

export type MatchResultResponse = {
  session_id: number
  payload: MatchResultPayload
  created_at: string
}

File: app/schemas/match.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class MatchSessionCreateRequest(BaseModel):
    client_user_id: int = Field(..., gt=0)


class MatchSessionResponse(BaseModel):
    id: int
    client_user_id: int
    status: str
    is_active: bool
    created_at: datetime
    updated_at: datetime


class MatchMessageCreateRequest(BaseModel):
    content: str = Field(..., min_length=1)


class MatchMessageResponse(BaseModel):
    id: int
    session_id: int
    role: str
    content: str
    created_at: datetime


class MatchResultResponse(BaseModel):
    session_id: int
    payload: dict[str, Any]
    created_at: datetime


class MatchSessionStatusUpdate(BaseModel):
    status: str